ui/hooks/usePrevious.ts

```diff
@@ -1,7 +1,7 @@
 import { useEffect, useRef } from 'react';
 
-export function usePrevious(value) {
-  const ref = useRef();
+export function usePrevious<T>(value: T): T | undefined {
+  const ref = useRef<T | undefined>();
```

Comment:
needed to update this since `*/test.tsx` file to be used in .tsx file properly

code that errored:
```
      jest
        .spyOn(usePreviousHooks, 'usePrevious')
        .mockImplementation(() => unapprovedTypedSignMsgV4.id);
```

error:
```
Argument of type '() => string' is not assignable to parameter of type '(value?: any) => undefined'.
  Type 'string' is not assignable to type 'undefined'.
```